openedx/core/djangoapps/credit/api/eligibility.py

```diff
@@ -225,6 +228,15 @@ def set_credit_requirement_status(username, course_key, req_namespace, req_name,
             )
 
     """
+    username = user.username
+    # Check whether user has credit enrollment. Only credit enrollments should be eligible for credit.
+    enrollment_mode, is_active = CourseEnrollment.enrollment_mode_for_user(user, course_key)
+    has_credit_enrollment = (CourseMode.is_credit_slug(enrollment_mode) and is_active)
```

Comment:
How about make sure the user has enrollment **not in `honor` or `audit` modes**?